From akpm@linux-foundation.org Wed Nov 21 15:08:10 2007
From: akpm@linux-foundation.org
Date: Wed, 21 Nov 2007 15:07:11 -0800
Subject: PCI: always export pci_scan_single_device
To: greg@kroah.com
Cc: akpm@linux-foundation.org, bunk@kernel.org, norsk5@yahoo.com
Message-ID: <200711212307.lALN7BHs013908@imap1.linux-foundation.org>


From: Adrian Bunk <bunk@kernel.org>

This patch fixes the following build error with CONFIG_HOTPLUG=n:

  MODPOST 2137 modules
ERROR: "pci_scan_single_device" [drivers/edac/i82875p_edac.ko] undefined!

Signed-off-by: Adrian Bunk <bunk@kernel.org>
Acked-by: Doug Thompson <norsk5@yahoo.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/probe.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -988,6 +988,7 @@ struct pci_dev *pci_scan_single_device(s
 
 	return dev;
 }
+EXPORT_SYMBOL(pci_scan_single_device);
 
 /**
  * pci_scan_slot - scan a PCI slot on a bus for devices.
@@ -1175,7 +1176,6 @@ EXPORT_SYMBOL(pci_add_new_bus);
 EXPORT_SYMBOL(pci_do_scan_bus);
 EXPORT_SYMBOL(pci_scan_slot);
 EXPORT_SYMBOL(pci_scan_bridge);
-EXPORT_SYMBOL(pci_scan_single_device);
 EXPORT_SYMBOL_GPL(pci_scan_child_bus);
 #endif
 
